### PR TITLE
perf: quince release support

### PR DESCRIPTION
**Pull request by luisfelipec95**
## Description

This PR adds support for the Open edx Quince release.

## Testing instructions

1. Create an environment with Quince release, you can use [Tutor](https://docs.tutor.overhang.io/index.html) or [TVM](https://github.com/eduNEXT/tvm)
2. Install the Django plugin eox-hooks, you can use steps in [Tutor documentation](https://docs.tutor.overhang.io/configuration.html#installing-extra-xblocks-and-requirements).
3. Run [test cases](https://docs.google.com/document/d/1f7LkXyxUbBcAdkDsWhtEzNfVNFFmV65enpKrEUUnirw/edit?usp=sharing) 

## Aditional information
[JIRA ISSUE DS-778](https://edunext.atlassian.net/jira/software/c/projects/DS/boards/40?selectedIssue=DS-778)

### CODE REVIEW

**Comment by bra-i-am** [comment]
hi @luisfelipec95, I hope you're doing well

I've been testing this PR (I'm just testing the functionality and haven't seen the code yet), but something is going wrong with the execution; did you follow the `EOX hooks test cases` to the letter, or did something different that we should be aware of?

Some examples:
1. image

2. And I'm not being able to unenroll from a course with any account

**Comment by luisfelipec95** [comment]
> hi @luisfelipec95, I hope you're doing well
> 
> I've been testing this PR (I'm just testing the functionality and haven't seen the code yet), but something is going wrong with the execution; did you follow the `EOX hooks test cases` to the letter, or did something different that we should be aware of?
> 
> Some examples:
> 
>     1. This error happened to me trying to run the [test case number 3](https://docs.google.com/document/d/1f7LkXyxUbBcAdkDsWhtEzNfVNFFmV65enpKrEUUnirw/edit#heading=h.oas0hz2ofaux)
> 
> 
> image
> 
>     2. And I'm not being able to unenroll from a course with any account trying to run the [test case number 6](https://docs.google.com/document/d/1f7LkXyxUbBcAdkDsWhtEzNfVNFFmV65enpKrEUUnirw/edit#heading=h.vm6mtsicj7jx)

Thanks, I already made a fix

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -11,21 +11,21 @@ jobs:
     strategy:
       max-parallel: 2
       matrix:
-        python-version: ["3.8", "3.10", "3.11"]
+        python-version: ["3.8"]
         django: ["32"]
```
Comment (bra-i-am) [inline .github/workflows/tests.yml:15]: According to [this](https://github.com/openedx/edx-platform/blob/76330b36b004f1636908ada962bbc75f4857eeb3/.github/workflows/unit-tests.yml#L20) and taking into account that Quince starts using Django 4.2, I suggest...

```suggestion
        django: ["32","42"]
```

File: eox_hooks/receivers.py [hunk 2]
```diff
@@ -26,7 +26,7 @@ def hooks_handler(sender, signal, **kwargs):
 
     trigger_event = None
     for uid in _get_dispatch_uid(signal.receivers):
-        if uid.startswith('eox-hooks:'):
+        if str(uid).startswith('eox-hooks:'):
```
Comment (bra-i-am) [inline eox_hooks/receivers.py:29]: Just because of curiosity... Is this the best way of solving the issue? From the beginning, why did this issue start appearing after 4 years?

File: .github/workflows/tests.yml [hunk 3]
```diff
@@ -11,21 +11,21 @@ jobs:
     strategy:
       max-parallel: 2
       matrix:
-        python-version: ["3.8", "3.10", "3.11"]
```
Comment (bra-i-am) [inline .github/workflows/tests.yml:14]: Why did you delete this Python versions?

**Comment by bra-i-am** [comment]
Thank you so much, @luisfelipec95, now the main functionality is working as expected.

Now, reviewing the code I have some comments:
1. Despite the `labeler` workflow is not failing, the Label is not being applied properly; I would recommend updating the dependency to `codelytv/pr-size-labeler@v1.7.0`
2. Remember to update the [constraints.txt](https://github.com/eduNEXT/eox-hooks/blob/master/requirements/constraints.txt), e.g, now django should be something like `Django<5.0` because edx-platform now uses [Django==4.2.8](https://github.com/openedx/edx-platform/blob/9ca3c694685b59168dd9956a50b89c089f490e32/requirements/edx/base.txt#L184)

File: eox_hooks/receivers.py (hunk 2 above)
Comment (MaferMazu) [reply eox_hooks/receivers.py:29]: Personally, the str() method doesn't bother me, but I think It is because I don't know a better way; if you (@luisfelipec95) find it, please go ahead. But it probably is a good idea to debug that part of the function to understand a little bit about what it should be that uid because, for the name, I thought it was a [UUID](https://docs.python.org/3/library/uuid.html), but in the log of Brayan, I read that the uid in at least of one iteration is an int.

By the way, @luisfelipec95, when you tried the flow that failed to @bra-i-am, did it fail in your env, too? I am starting to be paranoid about the environment. xD.

File: .github/workflows/tests.yml (hunk 3 above)
Comment (bra-i-am) [reply .github/workflows/tests.yml:14]: @MaferMazu, thanks a lot for your answer, now it is pretty clear to me

File: .github/workflows/tests.yml (hunk 1 above)
Comment (luisfelipec95) [reply .github/workflows/tests.yml]: Thank you very much!! I've already made the corrections.

**Comment by luisfelipec95** [comment]
> Thank you so much, @luisfelipec95, now the main functionality is working as expected.
> 
> Now, reviewing the code & docs I have some comments:
> 
>     1. Despite the `labeler` workflow is not failing, the Label is not being applied properly; I would recommend updating the dependency to `codelytv/pr-size-labeler@v1.7.0`
> 
>     2. Remember to update the [constraints.txt](https://github.com/eduNEXT/eox-hooks/blob/master/requirements/constraints.txt), e.g, now django should be something like `Django<5.0` because edx-platform now uses [Django==4.2.8](https://github.com/openedx/edx-platform/blob/9ca3c694685b59168dd9956a50b89c089f490e32/requirements/edx/base.txt#L184)
> 
>     3. And would be great if in the [test cases file](https://docs.google.com/document/d/1f7LkXyxUbBcAdkDsWhtEzNfVNFFmV65enpKrEUUnirw/edit) and in this PR you specify that is not possible to test the [Test Certificate Created](https://docs.google.com/document/d/1f7LkXyxUbBcAdkDsWhtEzNfVNFFmV65enpKrEUUnirw/edit#heading=h.oas0hz2ofaux) yet if is enable the learner-dashboard MFE ([here is the way to disable it](https://github.com/overhangio/tutor-mfe?tab=readme-ov-file#disabling-individual-mfes))

Thank you very much!! I've already made the corrections.

File: eox_hooks/receivers.py (hunk 2 above)
Comment (luisfelipec95) [reply eox_hooks/receivers.py:29]: I was able to replicate the error in my environment, which is why I could solve it easily; I didn't find another way. Thank you very much for the feedback.

**Review by bra-i-am — DISMISSED** [review]
LGTM!

**Review by MaferMazu — COMMENTED** [review]
Thanks for this, @luisfelipec95.
- [x] You update the chore requirements.
- [x]  Updated the Github actions.
- [x]  Update the requirements.
    - Sorry for my late review, but I think we need to modify the constraints. If, in the constraint, we have something like djangorestframework<3.13.0 or opened-events==4.1.1, we are forced to use that version instead of the version by default from quince ([restframework](https://github.com/openedx/edx-platform/blob/open-release/quince.master/requirements/edx/base.txt#L382) and [events](https://github.com/openedx/edx-platform/blob/open-release/quince.master/requirements/edx/base.txt#L780) version), at least in our GitHub tests. I recommend looking at those constraints and maintaining only the necessary ones, trying to avoid the super strict constraints == (unless required).
- [x]  Update the readme and other doc.
    - I think we need to update a little bit the format of the quince code block a (add lines between the content and the code-block because that info [doesn't appear correctly in the readme](https://github.com/eduNEXT/eox-hooks/blob/lfc/quince_release_support/README.rst)), if it makes sense maybe group the release with similar config, for example Palm, you can say Palm and Quince, and that's it.
- [x]  The test cases work.

And again, sorry for not giving my feedback before.

**Comment by luisfelipec95** [comment]
> Thanks for this, @luisfelipec95.
> 
>     * [x]  You update the chore requirements.
> 
>     * [x]   Updated the Github actions.
> 
>     * [ ]   Update the requirements.
>       
>       * Sorry for my late review, but I think we need to modify the constraints. If, in the constraint, we have something like djangorestframework<3.13.0 or opened-events==4.1.1, we are forced to use that version instead of the version by default from quince ([restframework](https://github.com/openedx/edx-platform/blob/open-release/quince.master/requirements/edx/base.txt#L382) and [events](https://github.com/openedx/edx-platform/blob/open-release/quince.master/requirements/edx/base.txt#L780) version), at least in our GitHub tests. I recommend looking at those constraints and maintaining only the necessary ones, trying to avoid the super strict constraints == (unless required).
> 
>     * [ ]   Update the readme and other doc.
>       
>       * I think we need to update a little bit the format of the quince code block a (add lines between the content and the code-block because that info [doesn't appear correctly in the readme](https://github.com/eduNEXT/eox-hooks/blob/lfc/quince_release_support/README.rst)), if it makes sense maybe group the release with similar config, for example Palm, you can say Palm and Quince, and that's it.
> 
>     * [x]   The test cases work.
> 
> 
> And again, sorry for not giving my feedback before.

Thank you very much for the feedback

**Review by MaferMazu — APPROVED** [review]
Thanks, @luisfelipec95. It looks good to me.

**Review by bra-i-am — APPROVED** [review]
Comments addressed & working as expected! 
Thanks ✨
